src/OneTimeTokenProvider.js

```diff
@@ -0,0 +1,43 @@
+import Connection from './Connection'
+
+export default class OneTimeTokenProvider {
+  constructor(publicAPIKey, options) {
+    this.publicAPIKey = publicAPIKey
+    this.options = options
+    this.token = null
+    this.timeout = null
+    this.marginToExpire = 0.05 // a margin to prevent unexpected expiration (5% of the time)
+  }
+
+  get() { return this.retrieve().then(() => this.getAndClearToken()) }
+
+  retrieve() {
```

Comment:
Thank you for your idea!
But considering [this comment I wrote](https://github.com/NEWROPE/scnnr-js/pull/13/files/c00230983f99354eb29d1d30cd9d8b4f0bf1690e#r202901321), I guess we can also reorganize methods as follows:

- `get` => `get` (no change)
- `retrieve` => `issue` (issue a token, and if a cache exists, reuse it)
- `issue` => `requestToken` (send a request to issue a token by calling the API anyway)

What do you think?

Reply:
I feel like it would be an improvement, as it reduces the amount of verbs with similar meanings, and the naming is a bit closer to the implementation.